src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -646,6 +647,12 @@ void TemporalMemory::compute(const SDR &activeColumns, bool learn,
 void TemporalMemory::compute(const SDR &activeColumns, bool learn) {
```

Comment:
@ctrl-z-9000-times I'd also prefer the TM.anomalyScore (const float&), to the TM::getAnomalyScore()

the problem is that with the former, I'll work if the user calls compute() but not if they call activateCells/Dendrites separately. Is this worth, or just use getAnomalyScore and implement a check with warning there?

Reply:
Would this issue go away if you put the anomaly code inside of `activateCells`?